src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -97,7 +97,7 @@ TargetWorkerSetNodeList(TargetWorkerSet targetWorkerSet, LOCKMODE lockMode)
 	{
 		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
 		if (targetWorkerSet == WORKERS_WITH_METADATA &&
-			!workerNode->hasMetadata)
+			(!workerNode->hasMetadata || !workerNode->metadataSynced))
```

Comment:
Otherwise some of the depended objects might not exist yet and this will error out.

Reply:
Should it be in a separate commit?

Reply:
done